docs/docs/loops-conditional-rendering.md

```diff
@@ -0,0 +1,228 @@
+  <li>Fred</li> /* key: "Fred" */
+</ul>
+```
+
+In the event that the element containing "Fred" needs to be removed, React will hide the element instead of removing the object entirely like so:
+
+```html
+<ul>
+  <li>John</li> /* key: "John" */
+  <li style={display: 'none'}>Fred</li> /* key: "Fred" */
+</ul>
+```
+
+#### Example 2
+
+Given the html below:
+
+```html
+<ul>
+  <li>John</li> /* key: "John" */
+  <li>Fred</li> /* key: "Fred" */
+</ul>
+```
+
+In the event that the element "John" needs to be removed, React will swap the text and hide the second element like so:
+
+```html
+<ul>
+  <li>Fred</li> /* key: "Fred" */
+  <li style={display: 'none'}>Fred</li> /* key: removed */
+</ul>
+```
+
+As your components begin managing their own `this.state`, including keys becomes increasingly important to the performance of your application.
+
+## Conditional Rendering
+
+With React you can render distinct components depending on the value of your data. This means no longer needing to manually modify the CSS classes of HTML elements. Instead, you would create distinct components that encapsulate behavior you need. Then, render only the specific component you need depending on the state of your application.
+
+This follows the concept of "separation of concerns" by allowing each component to manage how it should be rendered.
+
+### Component Variables
+
+Given the current state of your application you can assign components as the value of variables, then only render the needed component.
+
+In the example below we determine whether to allow the user to log in or log out based on whether or not they are already `loggedIn`. We assign to the variable `loginButton` one of the following components `<LoginButton /> and `<LogoutButton />`.
+
+```javascript
+LoginControl extends Component {
+  render() {
+    var loginButton;
+    if (loggedIn) {
+      loginButton = <LogoutButton />;
+    } else {
+      loginButton = <LoginButton />;
+    }
+
+    return (
+      <div>
+        {loginButton}
+      </div>
+    );
+  }
+}
+```
+
+### Prevent Component Rendering
+
+In some cases, you will not want a component to render. To prevent components from rendering, return `null` or `false` from the `render()` function.
+
+In the example below, a `<WarningBanner />` component will not be rendered within a `<Header /> component.
+
+```javascript
+class WarningBanner extends Component {
+  render () { return <div className="warning">Warning!</div>; }
+}
+class Header extends Component {
+  render() {
+    var displayWarning = false;
+    let warningBanner = null;
+    if (displayWarning) {
+      warningBanner = <WarningBanner />;
+    }
+    return (
+      <div>
+        {warningBanner}
+      </div>
+    );
+  }
+}
+```
+
+> A component `render()` function requires returning only one single component.
+
+### Rendering Multiple Components Using {}
+
+You can build collections of components and render them using the `{}` operator.
```

Comment:
It might be useful to first explain that `{}` lets you inline any JavaScript statement within JSX. This kind of implies that `{}` is used specifically for rendering collections of components.